src/main/scala/strawman/collection/mutable/HashSet.scala

```diff
@@ -2,35 +2,70 @@ package strawman.collection.mutable
 
 import strawman.collection.{IterableFactory, Iterator, MonoBuildable, PolyBuildable}
 
-import scala.{Any, Boolean, Option, Unit}
-import scala.Predef.???
+import scala.{Any, Boolean, Option, Serializable, SerialVersionUID, Unit}
 
-/** Mutable set backed by a hash trie */
-final class HashSet[A]
+/** This class implements mutable sets using a hashtable.
+  *
+  * @author  Matthias Zenger
+  * @author  Martin Odersky
+  * @version 2.0, 31/12/2006
+  * @since   1
+  * @see [[http://docs.scala-lang.org/overviews/collections/concrete-mutable-collection-classes.html#hash_tables "Scala's Collection Library overview"]]
+  * section on `Hash Tables` for more information.
+  *
+  * @define Coll `mutable.HashSet`
+  * @define coll mutable hash set
+  * @define mayNotTerminateInf
+  * @define willNotTerminateInf
+  */
+@SerialVersionUID(1L)
+final class HashSet[A](contents: FlatHashTable.Contents[A])
   extends Set[A]
     with SetLike[A, HashSet]
     with MonoBuildable[A, HashSet[A]]
     with PolyBuildable[A, HashSet]
-    with Builder[A, HashSet[A]] {
+    with FlatHashTable[A]
+    with Serializable {
 
-  def iterator(): Iterator[A] = ???
+  def this() = this(null)
+
+  override def iterator(): Iterator[A] = super[FlatHashTable].iterator
 
   def fromIterable[B](coll: strawman.collection.Iterable[B]): HashSet[B] =
     HashSet.fromIterable(coll)
   protected[this] def fromIterableWithSameElemType(coll: strawman.collection.Iterable[A]): HashSet[A] = fromIterable(coll)
-  protected[this] def newBuilderWithSameElemType: Builder[A, HashSet[A]] = new HashSet[A]
-  def newBuilder[E]: Builder[E, HashSet[E]] = new HashSet[E]
-  def result: HashSet[A] = this
+  protected[this] def newBuilderWithSameElemType: Builder[A, HashSet[A]] = newBuilder[A]
+  def newBuilder[E]: Builder[E, HashSet[E]] = HashSet.newBuilder[E]
+
+  def addInPlace(elem: A): this.type = {
+    addElem(elem)
```

Comment:
No, we don’t have it in every mutable collection type. Here `addElem` is a (`protected`) implementation detail.